### PR TITLE
Fix disposal search and analytics reports - Issue #14810

### PR DESCRIPTION
**Pull request by buddhika75**
### Prerequisites
- Access to pharmacy module with disposal bills in test data
- Different departments with disposal transactions
- Various disposal bill types (regular, cancelled, returns)

### Testing Scope
This fix addresses 6 key disposal report features that were showing incorrect data due to wrong bill type filtering.

### 1. Search Reports Testing

**Test: Search Disposal Bills**
- Navigate: Menu → Pharmacy → Disposal → Search Disposal Bills  
- Set date range covering known disposal periods
- Verify bills displayed are actual disposal bills (not pharmacy issue bills)
- Test filtering by bill number, department, and date ranges
- ✅ Expected: Only disposal bills appear in results

**Test: Search Disposal Items**  
- Navigate: Menu → Pharmacy → Disposal → Search Disposal Items
- Set date range and test various filters (item name, bill number, department)
- Verify items shown are from disposal bills only
- ✅ Expected: Only disposal bill items appear in results

### 2. Analytics Reports Testing

**Test: Unit Issue by Bill**
- Navigate: Menu → Pharmacy → Analytics → Disposal Reports → Unit Issue by bill
- Set date range and test department filtering
- Export to Excel and print functionality
- ✅ Expected: Shows disposal bills with correct values and departments

**Test: Unit Issue by Department**
- Navigate: Menu → Pharmacy → Analytics → Disposal Reports → Unit Issue by Department  
- Select From Department and date range
- Verify department totals reflect actual disposal amounts
- ✅ Expected: Accurate department-wise disposal totals

**Test: Unit Issue by Item (Batch)**
- Navigate: Menu → Pharmacy → Analytics → Disposal Reports → Unit Issue by Item (Batch)
- Test with different departments and item categories
- Verify batch numbers, purchase rates, retail rates display correctly
- Check quantity and value calculations
- ✅ Expected: Complete batch information with accurate financial data

**Test: Unit Issue by Item**
- Navigate: Menu → Pharmacy → Analytics → Disposal Reports → Unit Issue by Item
- Test department and category filtering
- Verify item-level quantities and net values
- ✅ Expected: Correct item disposal data without batch details

### 3. Cross-Validation Testing

**Verify Data Consistency**
- Compare search results with analytics reports for same date range
- Ensure bill counts and totals match across different report types
- Verify department names and values are consistent
- ✅ Expected: Consistent data across all disposal reports

**Test Edge Cases**
- Empty date ranges (should show no results)
- Date ranges with no disposal activity
- Department filtering with no matching data
- Very large date ranges for performance
- ✅ Expected: Graceful handling of edge cases

### 4. Performance Testing

**DTO Performance Validation**
- Generate reports with large datasets (6+ months)
- Monitor page load times and memory usage
- Test concurrent report generation
- ✅ Expected: Improved performance vs. previous entity-based queries

### 5. Regression Testing

**Ensure No Impact on Other Features**
- Test other pharmacy reports remain functional
- Verify non-disposal search features work correctly
- Check that regular pharmacy operations are unaffected
- ✅ Expected: No regression in existing functionality

### Technical Changes Summary
- Replaced BillType.PharmacyIssue with BillTypeAtomic.PHARMACY_DISPOSAL_ISSUE
- Converted entity-based queries to DTO-based for improved performance  
- Enhanced PharmacyItemPurchaseDTO with batch information support
- Updated all frontend pages to use new DTO properties
- Added comprehensive disposal reports wiki documentation

### Known Limitations
- Previous disposal report data inconsistencies are now resolved
- Some legacy entity references replaced with optimized DTO patterns
- Wiki documentation available at: https://github.com/hmislk/hmis/wiki/Disposal-Reports

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Richer Excel/PDF exports combining bills, bill items and fees, plus itemized sales and category breakdowns.
  * Disposal reporting now exposes item-level DTOs with batch, purchase/retail rates and margin for detailed exports.

* **Enhancements**
  * Pharmacy issue bill/item views simplified to DTO-backed columns (bill id, type, departments, date, gross/discount/net).
  * Unit reports show batch, rates, margin, quantity; cancellations/refunds handled more consistently.
  * Report parameters are redacted to protect PII.

* **Documentation**
  * Removed outdated wiki pages.

* **Chores**
  * Minor permissions list additions and autocomplete data source adjusted.

### CODE REVIEW

File: src/main/java/com/divudi/bean/pharmacy/ReportsTransfer.java [hunk 1]
```diff
@@ -1877,7 +1953,56 @@
...
+
+        if (category != null) {
+            jpql.append(" and bi.item.category = :cat");
+            m.put("cat", category);
+        }
+
+        jpql.append(" order by bi.item.name");
+
+        disposalIssueBillItemDtos = (List<PharmacyItemPurchaseDTO>) getBillFacade().findLightsByJpql(jpql.toString(), m, TemporalType.TIMESTAMP);
+
```
[the remark this answers is not in the document]
Comment (buddhika75) [reply src/main/java/com/divudi/bean/pharmacy/ReportsTransfer.java:2098]: @coderabitai - can you please recheck? I corrected and pushed this

[the remark this answers is not in the document]
Comment (buddhika75) [reply src/main/java/com/divudi/bean/pharmacy/ReportsTransfer.java]: @coderabbitai can you please check
